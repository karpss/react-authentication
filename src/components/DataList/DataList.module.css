.listContainer{
    display:flex;
    align-items: center;
    justify-content: center;
    flex-direction: column;
    margin-top: 2em;
}

.listSearch{
    background-color: transparent;
    border: 1px solid #010708;
    color: #010708;
    border-radius: 0.25rem;
    display: block;
    padding: 16px;
    margin-bottom: 4em;
    width: 250px;
    min-width: 150px;
}