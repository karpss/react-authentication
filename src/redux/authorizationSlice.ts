import { createSlice, PayloadAction } from "@reduxjs/toolkit";

export interface AuthorizationSliceState {
  token: string | null;
}

const initialState: AuthorizationSliceState = {
  token: null,
};

export const authorizationSlice = createSlice({
  name: "authorization",
  initialState,
  reducers: {
    tokenAssign: (state, action: PayloadAction<string>) => {
      state.token = action.payload;
    },
    tokenReset: (state) => {
      state.token = null;
    },
  },
});

export const { tokenAssign, tokenReset } = authorizationSlice.actions;

export const tokenSelect = (state: RootState) => state.authorization.token;

export default authorizationSlice.reducer;
