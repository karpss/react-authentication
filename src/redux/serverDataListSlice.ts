/* eslint-disable */
import { createSlice, PayloadAction } from "@reduxjs/toolkit";

export interface serverDataListSliceState {
    servers: { name: string; distance: number }[];
}

const initialState: serverDataListSliceState = {
    servers: []
}

export const serverDataListSlice = createSlice({
name: "serverDataList",
initialState,
reducers: {
    setServerDataList: (
      state,
      action: PayloadAction<Array<{ name: string; distance: number }>>
    ) => {
      state.servers = action.payload;
    },
  },


    
});

export const { setServerDataList } = serverDataListSlice.actions;

export const selectServerDataList = (state: RootState) => state.serverDataList.servers;

export default serverDataListSlice.reducer;