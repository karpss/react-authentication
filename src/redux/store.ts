/* eslint-disable */
import { configureStore } from "@reduxjs/toolkit";
import authorizationReducer from "./authorizationSlice";
import serverDataListReducer from "./serverDataListSlice";

const store = configureStore({
  reducer: {
    authorization: authorizationReducer,
    serverDataList: serverDataListReducer,
  },
});


export default store;
