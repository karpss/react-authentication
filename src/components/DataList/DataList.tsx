/* eslint-disable */
import React,{useState, useEffect} from 'react';
import {getServerDataList} from "../../api/api";
import {tokenSelect} from "../../redux/authorizationSlice";
import {useAppDispatch, useAppSelector} from "../../hooks/typeManagementHook";
import {setServerDataList, selectServerDataList} from "../../redux/serverDataListSlice";
import DataTable from '../DataTable/DataTable';

function DataList ()  {
const token = useAppSelector(tokenSelect);
const serverDataList = useAppSelector(selectServerDataList);
const dispatch = useAppDispatch();

const [search, setSearch] = useState<string>("");
const [dataListError, setDataListError] = useState<string>("");

useEffect(() => {
  getServerDataList(token)
  .then((res) => {
    if (!res.ok) {
      throw Error(
        "Error! Try again later"
      );
    }
    setDataListError("");
    return res.json();
  })
  .then((data) => dispatch(setServerDataList(data)))
  .catch((error) => setDataListError(error.message));

}, [dispatch, token]);








  return (
    <div>
      <input
          type="text"
          onChange={(event) => setSearch(event?.target.value)}
          placeholder="Search"
        />
        {dataListError ? (
        <p>{dataListError}</p>
      ) : (
        <DataTable serverDataList={serverDataList} search={search} />
      )}



    </div>
  )
}

export default DataList;