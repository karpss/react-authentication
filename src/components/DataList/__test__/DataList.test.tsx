/* eslint-disable */
import React from "react";
import { render } from "@testing-library/react";
import DataList from "../DataList";
import { Provider } from "react-redux";
import { store } from "../../../redux/store";

test("matches snapshot", () => {
  const { container } = render
  (<Provider store={store}>
  <DataList />
  </Provider>
  );
  expect(container.firstChild).toMatchSnapshot();
});
