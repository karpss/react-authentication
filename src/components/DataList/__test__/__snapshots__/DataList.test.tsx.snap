// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  class="listContainer"
>
  <input
    class="listSearch"
    placeholder="Search"
    type="text"
  />
  <div
    class="datatableContainer"
  >
    <p>
      Servers not found.
    </p>
  </div>
</div>
`;
